lib/awesome_print/formatters/ext/nokogiri_document_formatter.rb

```diff
@@ -0,0 +1,30 @@
+require_relative '../base_formatter'
+
+module AwesomePrint
+  module Formatters
+    class NokogiriDocumentFormatter < BaseFormatter
+
+      formatter_for :nokogiri_xml_document
+
+      def self.formattable?(object)
+        defined?(::Nokogiri) && object.is_a?(::Nokogiri::XML::Document)
+      end
+
+      def format(object)
+
+        xml = object.to_xml(indent: 2)
+        #
+        # Colorize tag, id/class name, and contents.
+        #
+        xml.gsub!(/(<)(\/?[A-Za-z1-9]+)/) { |tag| "#{$1}#{colorize($2, :keyword)}" }
+        xml.gsub!(/(id|class)="[^"]+"/i) { |id| colorize(id, :class) }
+        xml.gsub!(/>([^<]+)</) do |contents|
+          contents = colorize($1, :trueclass) if contents && !contents.empty?
+          ">#{contents}<"
+        end
```

Comment:
May be worth considering a Nokogiri helper for some of these features.

Depending on how much you want to look into this code it may also make sense to transfer the regex into named and documented constants to make it clearer what this code does.